packages/ui/src/components/token/TokenCustomView.tsx

```diff
@@ -0,0 +1,349 @@
+            : {}
+
+    const tokenAddresses = useRef(
+        Object.keys(tokens ?? {}).map((v) => v.toLowerCase())
+    ).current
+    const tokenSymbols = useRef(
+        Object.keys(tokens ?? {}).map((key) => tokens[key].symbol.toLowerCase())
+    ).current
+
+    const {
+        register,
+        handleSubmit,
+        setError,
+        setValue,
+
+        formState: { errors },
+    } = useForm<customTokenFormData>({
+        resolver: yupResolver(customTokenSchema),
+    })
+
+    const [isFirstIteration, setIsFirstIteration] = useState<boolean>(true)
+    const [isFirstLoading, setIsFirstLoading] = useState<boolean>(true)
+    // const [isCustomTokenEmpty, setIsCustomTokenEmpty] = useState<boolean>(false)
+    const [result, setResult] = useState<TokenResponse>({
+        address: "",
+        decimals: undefined,
+        logo: "",
+        name: "",
+        symbol: "",
+        type: "",
+    })
+    const [message, setMessage] = useState("")
+
+    const [isLoading, setIsLoading] = useState(false)
+
+    useEffect(() => {
+        setResult((prevState) => ({
+            ...prevState,
+            address: customTokenAddress,
+        }))
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [])
+
+    useEffect(() => {
+        setValue("tokenAddress", result.address)
+        setValue("tokenDecimals", result.decimals?.toString() ?? "")
+        setValue("tokenSymbol", result.symbol)
+    }, [result.address, result.decimals, result.symbol, setValue])
+
+    const onSubmit = handleSubmit(async (data: customTokenFormData) => {
+        try {
+            // Valid form data
+            if (
+                data.tokenAddress.length !== 42 ||
+                data.tokenAddress.substring(0, 2) !== "0x"
+            ) {
+                return setMessage("Enter a valid address.")
+            }
+
+            if (
+                !/^[a-zA-Z0-9_.$-]*$/.test(data.tokenSymbol) ||
+                data.tokenSymbol === ""
+            ) {
+                return setMessage("Enter valid symbol.")
+            }
+
+            if (
+                !data.tokenDecimals ||
+                isNaN(parseInt(data.tokenDecimals)) ||
+                data.tokenDecimals.length > 2
+            ) {
+                return setMessage("Enter valid decimals.")
+            }
+
+            const newToken = {
+                address: data.tokenAddress,
+                decimals: data.tokenDecimals,
+                logo: "",
+                name: data.tokenSymbol,
+                symbol: data.tokenSymbol,
+                type: "",
+            }
+            const tokenToAdd = result.symbol ? result : newToken
+
+            // populate symbol logo for custom token
+            searchTokenInAssetsList(tokenToAdd.symbol.toUpperCase()).then(
+                (res) => {
+                    const exactMatch = res.filter(
+                        (r) =>
+                            r.symbol.toLowerCase() ===
```

Comment:
There is a mix of syntax here. We're using .then and using a `try` `catch` clause. I would await the `searchTokenInAssetsList` instad of using the `.then` thing